src/main/kotlin/traffic_simulation/RoadNetwork.kt

```diff
@@ -2,33 +2,50 @@ package traffic_simulation
 
 class RoadNetwork(val capacity: Int) {
 
-    fun calculateDemand(vehicles: List<Vehicle>): Int {
-        var capacityDemand: Int = 0
+    fun calculateDemandFor24Hours(vehicles: List<Vehicle>): MutableMap<Int , Int> {
```

Comment:
Mir kam es als kleinste mögliche Änderung vor, um die bisherigen Funktionen des Programms zu erhalten. Du meinst man hätte besser z.B. für sowas wie `delayed` einen festen Boolean gesetzt - der nicht über die Funktionen bestimmt wird und als Output auch erstmal eine Fantasie-Liste gesetzt?

Finde das hätte die Zusammenhänge des Codes entfernt - ihn zerstückelt in unabhängige Einheiten und das wäre auch nicht so toll.